src/caterpillar.jl

```diff
@@ -0,0 +1,65 @@
+"""
+    RanefInfo
+
+Information on random effects conditional modes/means, variances, etc.
+
+Used for creating caterpillar plots
+"""
+struct RanefInfo{T<:AbstractFloat}
+    cnames::Vector{String}
+    levels::Vector
+    ranef::Matrix{T}
+    stddev::Matrix{T}
+end
+
+"""
+    ranefinfo(m::LinearMixedModel)
+
+Return a `NamedTuple{fnames(m), NTuple(k, RanefInfo)}` from model `m`
+"""
+function ranefinfo(m::LinearMixedModel{T}) where {T}
+    fn = fnames(m)
+    val = sizehint!(RanefInfo[], length(fn))
+    for (re, eff, cv) in zip(m.reterms, ranef(m), condVar(m))
+        push!(
+            val, 
+            RanefInfo(
+                re.cnames,
+                re.levels,
+                Matrix(adjoint(eff)),
+                Matrix(adjoint(dropdims(sqrt.(mapslices(diag, cv, dims=1:2)), dims=2))),
+            )
+        )
+    end
+    NamedTuple{fn}((val...,))
+end
+
+function caterpillar!(f::Figure, r::RanefInfo; orderby=1)
```

Comment:
it took me a few minutes to understand this, but `orderby` is just telling it which of the random effects to sort by and defaults to the first (so usually the intercept), correct?

Reply:
I added more documentation.